web/src/components/ChatController.tsx

```diff
@@ -54,7 +54,10 @@ const ChatController = ({ city, language }: ChatControllerProps): ReactElement =
     })
 
     if (data !== null) {
-      setData(messages => [...(messages ?? []), data])
+      setData({
+        typing: data.typing,
+        messages: data.messages,
+      })
```

Comment:
🔧 There should be no need for this:
```suggestion
      setData(data)
```

Reply:
Sure! It overrides the data we get from the `useLoadFromEndpoint` hook and persists it. Useful in this case as to avoid duplicated state.